src/test/java/com/squareup/javapoet/TypeSpecTest.java

```diff
@@ -1914,7 +1914,19 @@ private String toString(TypeSpec typeSpec) {
       CodeBlock.builder().add("$1L $2L", "a", "b", "c");
       fail();
     } catch (IllegalArgumentException expected) {
-      assertThat(expected).hasMessage("unused arguments: expected 2, received 3");
+      assertThat(expected).hasMessage("unused arguments are marked as zero: [1, 1, 0]");
+    }
+    try {
+      CodeBlock.builder().add("$1L $2L $1L", "a", "b", "c");
+      fail();
+    } catch (IllegalArgumentException expected) {
+      assertThat(expected).hasMessage("unused arguments are marked as zero: [2, 1, 0]");
```

Comment:
I think this messaging is a little confusing. Maybe just say: `'arguments are unused"` ?